packages/core/src/fetchers/gatherAutoFetchedReferentsFromIncludes.ts

```diff
@@ -0,0 +1,74 @@
+import { Entry, Asset, EntryCollection, EntrySkeletonType, UnresolvedLink } from 'contentful';
+import { DeepReference } from '@/deep-binding/deep-reference';
+export type MinimalEntryCollection = Pick<
+  EntryCollection<EntrySkeletonType, 'WITHOUT_LINK_RESOLUTION'>,
+  'items' | 'includes'
+>;
+
+/**
+ * Traverses deep-references and extracts referents (referent entities)
+ */
+export function gatherAutoFetchedReferentsFromIncludes(
+  deepReferences: DeepReference[],
+  entriesResponse: MinimalEntryCollection
+) {
+  const autofetchedReferentEntries: Entry[] = [];
+  const autofetchedReferentAssets: Asset[] = [];
+
+  for (const reference of deepReferences) {
+    const headEntry = entriesResponse.items.find(
+      (entry) => entry.sys.id === reference.headEntityId
+    );
+    if (!headEntry) {
+      throw new Error(
+        `LogicError: When resolving deep-references could not find reference-entry ${reference.entityId}`
+      );
+    }
+
+    const linkToReferent = headEntry.fields[reference.field] as UnresolvedLink<'Asset' | 'Entry'>;
+
+    if (!linkToReferent) {
+      debugger;
+      console.warn(
+        `fetchReferencedEntities():: link to referent not found within reference-field. Probably reference is simply not set.`
+      );
+      continue;
+    }
+
+    if (linkToReferent.sys.linkType === 'Entry') {
+      const referentAsset = entriesResponse.includes?.Entry?.find(
```

Comment:
Shouldn't this be `referentEntry`? Also since deep binding to entries is not supported yet do we need to handle this case?

In general as I mentioned before, it's better for unsupported cases or if the linked asset has been removed I think it's better to return undefined instead of throwing an error. This way the component will fallback to default value if present and the website won't crash.

Reply:
Let me know what you think :)

Reply:
DONE (as shown above)